src/inference/is.jl

```diff
@@ -1,60 +1,74 @@
 """
-    IS(n_particles::Int)
+    IS()
 
 Importance sampling algorithm.
 
 Note that this method is particle-based, and arrays of variables
 must be stored in a [`TArray`](@ref) object.
 
-Arguments:
-
-- `n_particles` is the number of particles to use.
-
 Usage:
 
 ```julia
-IS(1000)
+IS()
 ```
 
 Example:
 
 ```julia
 # Define a simple Normal model with unknown mean and variance.
 @model gdemo(x) = begin
-    s ~ InverseGamma(2, 3)
-    m ~ Normal(0, sqrt.(s))
+    s ~ InverseGamma(2,3)
+    m ~ Normal(0,sqrt.(s))
     x[1] ~ Normal(m, sqrt.(s))
     x[2] ~ Normal(m, sqrt.(s))
     return s, m
 end
 
-sample(gdemo([1.5, 2]), IS(1000))
+sample(gdemo([1.5, 2]), IS(), 1000)
 ```
 """
-mutable struct IS <: InferenceAlgorithm
-    n_particles ::  Int
-end
+struct IS{space} <: InferenceAlgorithm end
 
-function Sampler(alg::IS, s::Selector)
-    info = Dict{Symbol, Any}()
-    Sampler(alg, info, s)
+IS() = IS{()}()
+transition_type(::Sampler{<:IS}) = Transition
```

Comment:
At the moment I can't do this for the HMC samplers because the typing is not fully defined at the outset, because AHMC returns an additional `NamedTuple`. It'd be possible with a little elbow grease but it's more complexity than I'm willing to do at this stage. I've strongly typed the non-HMC samplers for now.

Reply:
Ok, we can take a look at this after. After the PR is merged, we can open issues for all the loose ends in the discussions here. I think we should address them before a major release.